pandas/core/frame.py

```diff
@@ -437,6 +437,9 @@ def __init__(
         columns: Optional[Axes] = None,
         dtype: Optional[Dtype] = None,
         copy: bool = False,
+        # TODO do we want to keep this as a keyword as well? (I think it can be handy)
+        # can we somehow make it a "private" keyword? (`_manager` ?)
+        manager: Optional[str] = None,
```

Comment:
I think i'd be ok with ```flags`` being an added keyword to the constructor (and you can then make manager a flag)

Reply:
If this is just for convenience (to easily change back and forth) I'd prefer a private method on DataFrame to change the manager (or return a new DataFrame with a new manager).

Reply:
The keyword right now makes it a bit more convenient to pick a specific one in the tests, or to compare both versions side by side. 

Eg instead of 

```
with pd.options_context("mode.data_manager", "block"):
    df = pd.DataFrame(...)
```

you can do 

```
df = pd.DataFrame(..., manager="block")
```

if you want eg a certain test to only run using a specific manager, regardless of the global setting.

I fully agree we should be careful with making this a public keyword, but I think that also for internal use eg in tests, it would be good to have a convenient way to do this. 

With a private method, you are thinking of a class method like `pd.DataFrame._construct_with_array_manager(...)` / `pd.DataFrame._construct_with_block_manager(..)` ?

Reply:
> With a private method, you are thinking of a class method like pd.DataFrame._construct_with_array_manager(...) / pd.DataFrame._construct_with_block_manager(..) ?

I was thinking `pd.DataFrame(...)._as_array_manager()`. But whatever is easiest for testing.

Reply:
Ah, OK. For testing that should also be fine (as long as it's not testing the constructor ;)). It will be less efficient as it's only converting to the other format after initial construction (which I do now anyway, but the goal is to fix that at some point of course), but for testing purposes that doesn't matter. 

Could also be a `pd.DataFrame(..)._as_manager("block"/"array")` ? 
Because we want to have both. Eg some test might be testing specifically aspects of block -based dataframe, so even when running the tests with global option to use array manager, that test should still use block manager.